### DISCUSSION TITLE
MDI Notepad example

### DISCUSSION THREAD

**Discussion by chunmingwang** [XusinboyBekchanov/VisualFBEditor#371] [Show and tell]
I place a MDI Notepad for preview.
Welcome your suggestions and improvements.

Todo: print & print setting...

MDINotepad Example for VisualFBEditor

![image](https://user-images.githubusercontent.com/35757455/192554394-3ffa0e57-ad10-4561-a785-614f663848e6.png)

Dark Mode

![image](https://user-images.githubusercontent.com/35757455/192554530-800b9890-789e-475d-b656-e399c91f1ba2.png)

Code Page

![image](https://user-images.githubusercontent.com/35757455/192557311-bac8b3e0-ec5f-49f4-9879-03e46e7290f1.png)

Find & Replace

![image](https://user-images.githubusercontent.com/35757455/192554744-562b915c-5102-4c25-8fa0-1fa42b7864c2.png)

Goto

![image](https://user-images.githubusercontent.com/35757455/192554822-91caa681-2c2e-4df3-a0bc-94028c16c588.png)

File Search

![image](https://user-images.githubusercontent.com/35757455/192554953-7e20e94b-d230-4164-a004-ef7204bb59ec.png)

File Sync

![image](https://user-images.githubusercontent.com/35757455/192555921-704de5eb-bb04-4f7a-affc-b3c84c56f387.png)

Hash

![image](https://user-images.githubusercontent.com/35757455/192555154-bec9edd0-cde6-4a15-bf27-9d46113f56d4.png)

**Comment by XusinboyBekchanov** [comment]
Very good! In this comment, I pointed out another improvement regarding the MDI Dark mode: https://github.com/XusinboyBekchanov/VisualFBEditor/discussions/358#discussioncomment-3543611

**Reply by chunmingwang** [reply]
thanks

**Comment by XusinboyBekchanov** [comment]
Here are the warning fixes:
[MDINotepad (2).zip](https://github.com/XusinboyBekchanov/VisualFBEditor/files/9483686/MDINotepad.2.zip)

**Reply by chunmingwang** [reply]
thanks

**Comment by chunmingwang** [comment]
Fix: memory leak causes crash
[MDINotepad (3).zip](https://github.com/XusinboyBekchanov/VisualFBEditor/files/9488615/MDINotepad.3.zip)

**Reply by XusinboyBekchanov** [reply]
Good. If you sent a pull request as an example, it would be nice.

**Reply by chunmingwang** [reply]
ok, i will

**Reply by XusinboyBekchanov** [reply]
With each change, in github you can see what has changed.

**Reply by chunmingwang** [reply]
noticed

**Reply by XusinboyBekchanov** [reply]
Added: All Files Filter to MDINotepad example Save Dialog:
https://github.com/XusinboyBekchanov/VisualFBEditor/commit/00ccaacacf9e264dccec9575fefc55a698e4df1d

**Comment by chunmingwang** [comment]
Issue: 
If open a file larger than 64K, cannot display TextBox on MIDChild, (at screenshort right side).
If new untitle and use file insert , then no issue. (at screenshort left side).

![image](https://user-images.githubusercontent.com/35757455/188471904-cb809622-7ba2-44a2-9fa4-92bc471e642c.png)

File insert funtion on menu:

![image](https://user-images.githubusercontent.com/35757455/188471791-487e5058-6be9-40f6-ae61-e90b1e4e2827.png)

Please help to check.

**Comment by XusinboyBekchanov** [comment]
You should look at this value:
![image](https://user-images.githubusercontent.com/32607344/188484909-5d16e91d-531e-4d46-9bbf-aff9a03e0ee5.png)

**Reply by XusinboyBekchanov** [reply]
Here you can read:
https://docs.microsoft.com/en-us/windows/win32/controls/em-limittext

**Reply by XusinboyBekchanov** [reply]
Give us an example of a large file. I'll try to fix it.

**Reply by chunmingwang** [reply]
> Give us an example of a large file. I'll try to fix it.

[eng-utf8bom.txt](https://github.com/XusinboyBekchanov/VisualFBEditor/files/9492365/eng-utf8bom.txt)
[eng-ascii.txt](https://github.com/XusinboyBekchanov/VisualFBEditor/files/9492366/eng-ascii.txt)

**Reply by chunmingwang** [reply]
> Here you can read: https://docs.microsoft.com/en-us/windows/win32/controls/em-limittext

Thanks, i don't changed the MaxLength of TextBox,
if MDIChild is not show, then set the Text for TextBox, can't display if text size larger than 64K.
if MDIChild already shown, set the text size over 64K is work.

**Reply by chunmingwang** [reply]
I Open and File Insert using the same file.

![1662422917053](https://user-images.githubusercontent.com/35757455/188522449-53a87a18-9335-48ba-9776-dcd78ee2aa0c.png)

**Reply by XusinboyBekchanov** [reply]
Fixed: Opening file with text size larger than 64K in MDINotepad example:
https://github.com/XusinboyBekchanov/VisualFBEditor/commit/b6e34d4857185cadc2f27e81cc5b6bcbc7b30a77

So it will open.

**Reply by chunmingwang** [reply]
it works! thank you.

**Comment by hustbeef** [comment]
Why not try RichTextbox? It isinteresting if the RichTextbox could save and load as html file.

**Reply by XusinboyBekchanov** [reply]
RichTextBox can also be used. No, it can only save and read in RTF format.

**Reply by chunmingwang** [reply]
> Why not try RichTextbox? It isinteresting if the RichTextbox could save and load as html file.

I'm not familiar with RichTextBox

**Comment by chunmingwang** [comment]
Issue found:

I have changed the icon to a new style,
However, the Menu Editor and Menu display on Form still display the icons of VisualFBEeditor.
The ToolBar Editor and ToolBar on Form is normal.
Please help check.

![image](https://user-images.githubusercontent.com/35757455/188603772-5f33cbbf-1473-4500-a15f-06c695322f09.png)

Run time screenshot

![image](https://user-images.githubusercontent.com/35757455/188606233-1aa2fb48-38e3-4a7c-948e-79416c5085c5.png)

**Reply by XusinboyBekchanov** [reply]
Fixed: Designing Menu Item Images:
https://github.com/XusinboyBekchanov/VisualFBEditor/commit/378905218bc47e75e90a9d32ad60decc36672ed0

Fixed: Image and ImageKey property of MenuItem:
https://github.com/XusinboyBekchanov/MyFbFramework/commit/bfae78642e0e1ea19f417ff7ccc2cfc61ffc8094

**Reply by chunmingwang** [reply]
Thank you, works!

![image](https://user-images.githubusercontent.com/35757455/188653195-fac9370d-3087-4ee1-8cd2-d1006cffdae7.png)

**Reply by chunmingwang** [reply]
Issue found on OpenFileDialog:
1, Can't changed the property of MultiSelect on Form Editor.

![image](https://user-images.githubusercontent.com/35757455/189054856-db7b5d33-2c78-43f0-8722-b41f99e754c5.png)

2, MultiSelect is old style on run time.

```
OpenFileDialog1.MultiSelect = True 
OpenFileDialog1.Filter = "All Files (*.*)|*.*"
```
![image](https://user-images.githubusercontent.com/35757455/189055759-b9f11c88-4979-45d4-9951-829e8a6de8c2.png)

**Reply by XusinboyBekchanov** [reply]
Added: Added: Ability to change properties of OpenFileDialog and SaveFileDialog through the Properties window:
https://github.com/XusinboyBekchanov/MyFbFramework/commit/f59f8650f8232f38f5953fa9188ce2809b8fe1f5

And Old style also fixed.

**Reply by chunmingwang** [reply]
Test, everything is normal.
Thank you.

**Reply by chunmingwang** [reply]
New issue, both OpenFileDialog and SaveFileDialog are not displayed at the specified location on Form Editor.
![1662629316604](https://user-images.githubusercontent.com/35757455/189087056-5ebc455e-1e00-42fb-a1fd-02a682737ca9.png)

**Reply by XusinboyBekchanov** [reply]
Added: Ability to change properties of FolderBrowserDialog, FontDialog and ColorDialog through the Properties window, Fixed: Regress on Design:
https://github.com/XusinboyBekchanov/MyFbFramework/commit/8b0b3f2efc407d51c983cc8d6dad22368038ea3b

Fixed: Updating Code Editor when Move components on Designer:
https://github.com/XusinboyBekchanov/VisualFBEditor/commit/efb913f8aca8e1c3929b6fe70ef96525fa3a72c2

**Reply by chunmingwang** [reply]
Thank you.

**Comment by chunmingwang** [comment]
I wish everyone a Happy Mid Autumn Festival!

**Reply by XusinboyBekchanov** [reply]
Congratulations to you too.

**Comment by chunmingwang** [comment]
Add Hash, FileSearch & FileSync for MDINotepad Example
![1664322229680](https://user-images.githubusercontent.com/35757455/192656578-ef3769be-97b9-48ca-8ab2-3912852247c2.png)

They are also independent projects on MDINotepad.vfs session
![image](https://user-images.githubusercontent.com/35757455/192656801-08810a85-76b4-4f59-bc7b-10c3ee88b660.png)

**Reply by XusinboyBekchanov** [reply]
Good examples. I have merged the pull request.
